### PR TITLE
support 256 colors in print_with_color and env variables

### PR DESCRIPTION
**Pull request by KristofferC**
image

Now people can have their orange warning messages:

image

### CODE REVIEW

File: base/client.jl [hunk 1]
```diff
@@ -16,10 +16,14 @@ const text_colors = AnyDict(
     :bold    => "\033[1m",
 )
 
+for i in 0:256
```
Comment (tkelman) [inline base/client.jl]: isn't that 257 colors?

Comment (KristofferC) [inline base/client.jl]: lol yeah. Should be 0:255...

Comment (KristofferC) [inline base/client.jl]: fixed

Comment (StefanKarpinski) [inline base/client.jl]: I'm going to start calling you Tony "Eagle Eye" Kelman, a well-earned nickname.

**Comment by KristofferC** [comment]
This should be fairly uncontroversial since it doesn't change any old behavior. Anyone opposed for merging?

File: base/util.jl [hunk 2]
```diff
@@ -299,8 +299,9 @@ end
 
 ## printing with color ##
 
-function with_output_color(f::Function, color::Symbol, io::IO, args...)
+function with_output_color(f::Function, color::Union{Int, Symbol}, io::IO, args...)
     buf = IOBuffer()
+
```
Comment (tkelman) [inline base/util.jl]: this newline seems unnecessary

**Review by tkelman — APPROVED** [review]

File: base/util.jl (hunk 2 above)
Comment (KristofferC) [inline base/util.jl]: gone

**Comment by KristofferC** [comment]
Added a NEWS entry so maybe someone could look through that.

**Pull request merged**